pylint/checkers/classes/class_checker.py

```diff
@@ -880,6 +887,39 @@ def visit_classdef(self, node: nodes.ClassDef) -> None:
         self._check_proper_bases(node)
         self._check_typing_final(node)
         self._check_consistent_mro(node)
+        self._check_declare_non_slot(node)
+
+    def _check_declare_non_slot(self, node: nodes.ClassDef) -> None:
+        if not self._has_valid_slots(node):
+            return
+
+        slot_names = self._get_classdef_slots_names(node)
```

Comment:
`_get_classdef_slots_names` also calls `_has_valid_slots`.

Can we just return early if `_get_classdef_slots_names` returns `[]`?

Reply:
Good question... This got me thinking more about this implementation and I now think it has issues. For example, what to do when `__slots__` is present but empty? That's permitted if the class takes part in multiple inheritance. I also think it's unclear in my implementation whether I mean to just check for the presence of `__slots__` or to check that it is also a valid `__slots__` value. I'll need to think about this some more... :thinking: